src/Strategy/Python/Poetry.hs

```diff
@@ -180,21 +187,68 @@ analyze PoetryProject{pyProjectToml, poetryLock} = do
 
 -- | Use a `pyproject.toml` to set the direct dependencies of a graph created from `poetry.lock`.
 setGraphDirectsFromPyproject :: Graphing Dependency -> PyProject -> Graphing Dependency
-setGraphDirectsFromPyproject graph pyproject = Graphing.promoteToDirect isDirect graph
+setGraphDirectsFromPyproject graph pyproject = graph'
   where
+    graph' :: Graphing Dependency
+    graph' = hydrateDepEnvs $ Graphing.promoteToDirect isDirect graph
```

Comment:
[nit] Prefer a more specific name here, like `envHydratedGraph`. nbd since it's a very local name though.